src/Cake.Common/Tools/VSWhere/Product/VSWhereProductSettings.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Cake.Common.Tools.VSWhere.Product
+{
+    /// <summary>
+    /// Contains settings used by <see cref="VSWhereProduct"/>.
+    /// </summary>
+    public sealed class VSWhereProductSettings : VSWhereSettings
+    {
+        /// <summary>
+        /// Gets or sets the products to find. Defaults to Community, Professional, and Enterprise. Specify "*" by itself to search all product instances installed.
+        /// </summary>
+        public string Products { get; set; }
```

Comment:
This will currently be ignored if set see https://github.com/cake-build/cake/pull/1520/files#diff-96c402755256a225c5f4eb14a5f2586bR241